examples/stream.cs

```
@@ -0,0 +1,638 @@
+﻿namespace ZelloStream
+{
+    using System;
+    using System.IO;
+    using System.Collections.Generic;
+    using System.Net.WebSockets;
+    using System.Threading;
+    using System.Text.Json;
+    using System.Buffers.Binary;
+    using Microsoft.Extensions.Configuration;
+
+    // https://tools.ietf.org/html/rfc7845
+    // https://tools.ietf.org/html/rfc3533
+    class OpusFileStream : IDisposable {
+
+        Stream opusFile;
+        byte[] segmentSizes;
+        byte segmentIdx;
+        byte segmentsCount;
+        UInt32 sequenceNumber;
+        public UInt32 sampleRate { get; private set; }
+        int opusHeadersCount;
+        public int packetDurationMs { get; private set; }
+        public int framesPerPacket { get; private set; }
+        List<byte[]> savedPackets;
+
+        public OpusFileStream(string filename) {
+            try {
+                this.opusFile = File.OpenRead(filename);
+            } catch {
+                throw new Exception("Failed opening " + filename);

```

original exception will be lost, need to wrap

---

Added handling of the original exception